cmd/node/node.go

```diff
@@ -96,15 +95,23 @@ var Cmd = &cobra.Command{
 	Short: "start node",
 	Run: func(cmd *cobra.Command, args []string) {
 		app := NewSpacemeshApp()
-		defer app.Cleanup(cmd, args)
-
-		err := app.Initialize(cmd, args)
+		starter := func() error {
+			if err := app.Initialize(cmd, args); err != nil {
+				log.With().Error("Failed to initialize node.", log.Err(err))
+				return err
+			}
+			// This blocks until the context is finished
```

Comment:
```suggestion
			// This blocks until the context is finished or until an error is produced
```

Reply:
Are you sure this won't print a misleading error message if the app exits cleanly?

Reply:
node exits without error only if ctx is terminated